mautic/mautic
# DPMMA-2004 Campaign weekday and hour email metrics

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description
This PR introduces new campaign email metrics that display performance data by weekdays and hours. 
![image](https://github.com/user-attachments/assets/8b8fc95d-062f-492c-99d9-90b1617a6873)






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new campaign or use an existing one with email actions.
3. Add contacts to the campaign and ensure some emails are sent, opened, and clicked.
4. Navigate to the campaign details page.
5. Look for the new "Email Metrics" tab.
6. Click on the "Email Metrics" tab to view the charts.
7. Verify that the weekday chart shows email metrics (sent, read, clicked) for each day of the week.
8. Check that the hourly chart displays email engagement data (sent, read, clicked) across 24 hours.
9. Test the date range selector to view metrics for different time periods.
10. Change timezone in Mautic settings and confirm that the metrics update accordingly.
